[docs] Mention SWC in TypeScript documentation. (#33801)
vercel/next.js: docs/basic-features/typescript.md

You can also provide a relative path to a tsconfig.json file by setting `typescript.tsconfigPath` prop inside your `next.config.js` file.

> Next.js will by default use [Rust SWC](https://nextjs.org/docs/advanced-features/compiler) to compile TypeScript and TSX, resulting in a much faster build time.

Reviewer: ```suggestion
Starting in `v12.0.0`, Next.js uses [SWC](https://nextjs.org/docs/advanced-features/compiler) by default to compile TypeScript and TSX for faster builds.
```

> > Next.js will use Babel to handle TypeScript if `.babelrc` is present, this has some [caveats](https://babeljs.io/docs/en/babel-plugin-transform-typescript#caveats), and some [compiler options are handled differently](https://babeljs.io/docs/en/babel-plugin-transform-typescript#typescript-compiler-options).

Reviewer: ```suggestion
> Next.js will use Babel to handle TypeScript if `.babelrc` is present. This has some [caveats](https://babeljs.io/docs/en/babel-plugin-transform-typescript#caveats) and some [compiler options are handled differently](https://babeljs.io/docs/en/babel-plugin-transform-typescript#typescript-compiler-options).
```
